csharp/src/Google.Protobuf.Test/CodedInputStreamAsyncTest.cs

```diff
@@ -0,0 +1,502 @@
+#region Copyright notice and license
```

Comment:
Not really... This would require `ReadRawVarint32` to be an `async` method and return `Task<uint>`.
`async` means: use continuations semantics (you can use `await` only in `async` methods).